### PR TITLE
Feature/multi tenancy check scripts

### PR DESCRIPTION
**Pull request by bugalot**
The only interesting stuff to review related to TAO-5342 is the `scripts/tools/GenerateTranslationBundles.php` script. Other things are related to submarine requests from devops.

### CODE REVIEW

File: install/utils/class.System.php [hunk 1]
```diff
@@ -57,8 +57,10 @@ public static function getInfos(){
      * 
 	 * @return boolean
 	 */
-	public static function isTAOInstalled(){
-		$config = dirname(__FILE__).'/../../../config/generis.conf.php';
+	public static function isTAOInstalled($path = '')
+    {
+        $path = (empty($path)) ? dirname(__FILE__) . '/../../../config' : rtrim($path, '/\\');
```
Comment (antoinerobin) [inline install/utils/class.System.php]: you could use `__DIR__` instead of `dirname(__FILE__)`

File: install/utils/class.System.php [hunk 2]
```diff
@@ -67,16 +69,17 @@ public static function isTAOInstalled(){
      *
      * @return boolean
      */
-    public static function isTAOUpToDate()
+    public static function isTAOUpToDate($path = '')
     {
-        $generisConf = dirname(__FILE__).'/../../../config/generis.conf.php';
+        $path = (empty($path)) ? dirname(__FILE__) . '/../../../config' : rtrim($path, '/\\');
```
Comment (antoinerobin) [inline install/utils/class.System.php]: same here

File: install/utils/class.System.php [hunk 3]
```diff
@@ -86,7 +89,7 @@ public static function isTAOUpToDate()
         $extIterator = is_array($conf) ? $conf : $conf->getConfig();
 
         foreach ($extIterator as $extName => $ext) {
-            $manifestPath = dirname(__FILE__).'/../../../' . $extName . '/manifest.php';
+            $manifestPath = dirname(__FILE__) . "/../../../${extName}/manifest.php";
```
Comment (antoinerobin) [inline install/utils/class.System.php]: same here

File: scripts/tools/IsInstalled.php [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+ * the path to the configuration folder. In case of 'p' is not given,
+ * the script will search for configuration in INSTALL_PATH/config.
+ * 
+ * An optional argument 'v' enables verbose mode.
+ * 
+ * Example usage:
+ * sudo -u www-data php tao/scripts/tools/IsInstalled.php
+ * sudo -u www-data php tao/scripts/tools/IsInstalled.php -p ./config -v
+ */
+require(dirname(__FILE__) . '/../../../vendor/autoload.php');
```
Comment (antoinerobin) [inline scripts/tools/IsInstalled.php]: same here

File: scripts/tools/IsInstalled.php [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+$options = getopt('p:v::');
+$path = (!isset($options['p'])) ? '' : $options['p'];
```
Comment (antoinerobin) [inline scripts/tools/IsInstalled.php]: why p is required if you can have it empty or even not set ?

File: scripts/tools/IsInstalled.php [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+$isInstalled = \tao_install_utils_System::isTAOInstalled($path);
+$exitCode = ($isInstalled) ? 0 : 128;
+
+if (isset($options['v'])) {
```
Comment (antoinerobin) [inline scripts/tools/IsInstalled.php]: v is optionnal or do not accept value ?

File: scripts/tools/IsUpToDate.php [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+ * the path to the configuration folder. In case of 'p' is not given,
+ * the script will search for configuration in INSTALL_PATH/config.
+ * 
+ * An optional argument 'v' enables verbose mode.
+ * 
+ * Example usage:
+ * sudo -u www-data php tao/scripts/tools/IsUpToDate.php
+ * sudo -u www-data php tao/scripts/tools/IsUpToDate.php -p ./config -v
+ */
+require(dirname(__FILE__) . '/../../../vendor/autoload.php');
```
Comment (antoinerobin) [inline scripts/tools/IsUpToDate.php]: use `__DIR__`

File: scripts/tools/IsUpToDate.php [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+
+$options = getopt('p:v::');
```
Comment (antoinerobin) [inline scripts/tools/IsUpToDate.php]: are you sure that p is required and v optional ?

Comment (bugalot) [reply scripts/tools/IsUpToDate.php]: did you give it a try ? :hankey:

File: scripts/tools/IsInstalled.php (hunk 6 above)
Comment (bugalot) [reply scripts/tools/IsInstalled.php]: Did you give it a try :hankey: ?

**Pull request merged**